ToolBox/IBDownloader/Program.cs

```diff
@@ -0,0 +1,151 @@
+using QuantConnect.Configuration;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+
+namespace QuantConnect.ToolBox.IBDownloader
+{
+    class Program
+    {
+        /// <summary>
+        /// Primary entry point to the program
+        /// </summary>
+        static void Main(string[] args)
+        {
+            if (args.Length != 5)
+            {
+                Console.WriteLine("Usage: IBDownloader SECURITY_TYPE SYMBOLS RESOLUTION FROMDATE TODATE");
+                Console.WriteLine("SECURITY_TYPE = eg Equity/Forex");
+                Console.WriteLine("SYMBOLS = eg EURUSD,USDJPY");
+                Console.WriteLine("RESOLUTION = Second/Minute/Hour/Daily/All");
+                Console.WriteLine("FROMDATE = yyyymmdd");
+                Console.WriteLine("TODATE = yyyymmdd");
+                Environment.Exit(1);
+            }
+            try
+            {
+                // Load settings from command line
+
+                SecurityType securityType;
+                if (!Enum.TryParse(args[0], out securityType)) { throw new ApplicationException("Can't parse securityType "+ args[0]);}
+
+                if (securityType != SecurityType.Equity && securityType != SecurityType.Forex) { throw new ApplicationException("SECURITY_TYPE = eg Equity/Forex"); }
+
+                var tickers = args[1].Split(',');
+                var allResolutions = args[2].ToLower() == "all";
+                var resolution = allResolutions ? Resolution.Second : (Resolution)Enum.Parse(typeof(Resolution), args[2]);
+                var startDate = DateTime.ParseExact(args[3], "yyyyMMdd", CultureInfo.InvariantCulture).ConvertToUtc(TimeZones.NewYork);
+                var endDate = DateTime.ParseExact(args[4], "yyyyMMdd", CultureInfo.InvariantCulture).ConvertToUtc(TimeZones.NewYork);
+
+                
+                //fix end date 
+                endDate = new DateTime(Math.Min(endDate.Ticks, DateTime.Now.AddDays(-1).Ticks));
+
+                // Max number of histoy days
+                int maxDays = 1;
+                if (!allResolutions)
+                {
+                    switch (resolution)
+                    {
+                        case Resolution.Daily:
+                            maxDays = 365;
+                            break;
+                        case Resolution.Hour:
+                            maxDays = 30;
+                            break;
+                        case Resolution.Minute:
+                            maxDays = 10;
+                            break;
+                    }
+                }
+
+                // Load settings from config.json
+                var dataDirectory = Config.Get("data-folder", "../../../Data");
+
+                // Create IB Broker Gateway Runner
+                InteractiveBrokersGatewayRunner.StartFromConfiguration();
+
+                // Create an instance of the downloader
+                const string market = Market.FXCM; // new market for IB ?
```

Comment:
This is the first downloader project supporting multiple asset types: maybe the market should be a command line argument (in addition to security type)?